[CI] DocsClientYamlTestSuiteIT test {yaml=reference/mapping/types/boolean/synthetic-source-boolean-example} failing

Release tests are failing for the synthetic source mapping type. Is it behind a feature flag?

**Build scan:**
https://gradle-enterprise.elastic.co/s/on6j2n454vgqw/tests/:docs:yamlRestTest/org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT/test%20%7Byaml=reference%2Fmapping%2Ftypes%2Fboolean%2Fsynthetic-source-boolean-example%7D

**Reproduction line:**
`./gradlew ':docs:yamlRestTest' --tests "org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT.test {yaml=reference/mapping/types/boolean/synthetic-source-boolean-example}" -Dtests.seed=A1A56C74C0A32F87 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=is-IS -Dtests.timezone=America/Bahia_Banderas -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.smoketest.DocsClientYamlTestSuiteIT&tests.test=test%20%7Byaml%3Dreference/mapping/types/boolean/synthetic-source-boolean-example%7D

**Failure excerpt:**
```
java.lang.AssertionError: Failure at [reference/mapping/types/boolean:175]: expected [2xx] status code but api [raw[method=PUT path=idx]] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"unknown parameter [synthetic] on mapper [_source] of type [null]","stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping: unknown parameter [synthetic] on mapper [_source] of type [null]","caused_by":{"type":"mapper_parsing_exception","reason":"unknown parameter [synthetic] on mapper [_source] of type [null]","stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:372)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\nCaused by: org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\t... 22 more\n"},"status":400}]

  at __randomizedtesting.SeedInfo.seed([A1A56C74C0A32F87:29F153AE6E5F427F]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:510)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
  at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(null:-1)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [raw[method=PUT path=idx]] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"unknown parameter [synthetic] on mapper [_source] of type [null]","stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping: unknown parameter [synthetic] on mapper [_source] of type [null]","caused_by":{"type":"mapper_parsing_exception","reason":"unknown parameter [synthetic] on mapper [_source] of type [null]","stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\n"},"stack_trace":"org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:372)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:347)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1233)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$3(MetadataCreateIndexService.java:432)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:671)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:430)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:544)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:392)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:399)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:484)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:929)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:899)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:249)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:157)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:709)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:260)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat java.base/java.lang.Thread.run(Thread.java:833)\nCaused by: org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [synthetic] on mapper [_source] of type [null]\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1293)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MetadataFieldMapper$ConfigurableTypeParser.parse(MetadataFieldMapper.java:102)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:118)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:94)\n\tat org.elasticsearch.server@8.4.0/org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:370)\n\t... 22 more\n"},"status":400}]

    at org.junit.Assert.fail(Assert.java:88)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:373)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:499)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:479)
    at jdk.internal.reflect.GeneratedMethodAccessor14.invoke(null:-1)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:568)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/docs/build.gradle
+++ b/docs/build.gradle
@@ -1,4 +1,4 @@
-import org.elasticsearch.gradle.VersionProperties
+import org.elasticsearch.gradle.Version
 import org.elasticsearch.gradle.internal.info.BuildParams
 
 import static org.elasticsearch.gradle.testclusters.TestDistribution.DEFAULT
@@ -91,6 +91,8 @@ testClusters.matching { it.name == "yamlRestTest"}.configureEach {
 
   // TODO: remove this once cname is prepended to transport.publish_address by default in 8.0
   systemProperty 'es.transport.cname_in_publish_address', 'true'
+
+  requiresFeature 'es.index_mode_feature_flag_registered', Version.fromString("8.0.0")
 
   // build the cluster with all plugins
   project.rootProject.subprojects.findAll { it.parent.path == ':plugins' }.each { subproj ->

